docker/build-env-release.Dockerfile

```diff
@@ -0,0 +1,137 @@
+    tar -xzf mpich-${MPICH_VERSION}.tar.gz && \
+    cd mpich-${MPICH_VERSION} && \
+    ./configure \
+      --disable-fortran \
+      --prefix=$MPICH_PATH && \
+    make install -j$(nproc) && \
+    rm -rf /root/mpich-${MPICH_VERSION}.tar.gz /root/mpich-${MPICH_VERSION}
+
+# Install MKL
+ARG MKL_VERSION=2020.0-088
+RUN wget -qO - https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB 2>/dev/null | apt-key add - && \
+    apt-add-repository 'deb https://apt.repos.intel.com/mkl all main' && \
+    apt-get install -y -qq --no-install-recommends intel-mkl-64bit-${MKL_VERSION} && \
+    rm -rf /var/lib/apt/lists/* && \
+    echo "/opt/intel/lib/intel64\n/opt/intel/compilers_and_libraries/linux/mkl/lib/intel64" >> /etc/ld.so.conf.d/intel.conf && \
+    ldconfig
+
+# Install Boost
+ARG BOOST_MAJOR=1
+ARG BOOST_MINOR=72
+ARG BOOST_PATCH=0
+ARG BOOST_PATH=/usr/local/boost
+
+RUN wget -q https://dl.bintray.com/boostorg/release/${BOOST_MAJOR}.${BOOST_MINOR}.${BOOST_PATCH}/source/boost_${BOOST_MAJOR}_${BOOST_MINOR}_${BOOST_PATCH}.tar.gz -O boost.tar.gz && \
+    tar -xzf boost.tar.gz && \
+    cd boost_${BOOST_MAJOR}_${BOOST_MINOR}_${BOOST_PATCH} && \
+    ./bootstrap.sh --prefix=$BOOST_PATH && \
+    ./b2 -j$(nproc) debug-symbols=on install && \
+    rm -rf /root/boost.tar.gz /root/boost_${BOOST_MAJOR}_${BOOST_MINOR}_${BOOST_PATCH}
+
+# Install hwloc
+ARG HWLOC_MAJOR=2
+ARG HWLOC_MINOR=2
+ARG HWLOC_PATCH=0
+ARG HWLOC_PATH=/usr/local/hwloc
+
+RUN wget -q https://download.open-mpi.org/release/hwloc/v${HWLOC_MAJOR}.${HWLOC_MINOR}/hwloc-${HWLOC_MAJOR}.${HWLOC_MINOR}.${HWLOC_PATCH}.tar.gz -O hwloc.tar.gz && \
+    tar -xzf hwloc.tar.gz && \
+    cd hwloc-${HWLOC_MAJOR}.${HWLOC_MINOR}.${HWLOC_PATCH} && \
+    ./configure --prefix=$HWLOC_PATH && \
+    make -j$(nproc) install && \
+    rm -rf /root/hwloc.tar.gz /root/hwloc-${HWLOC_MAJOR}.${HWLOC_MINOR}.${HWLOC_PATCH}
+
+# Install tcmalloc (their version tagging is a bit inconsistent; patch version is not always included)
+ARG GPERFTOOLS_VERSION=2.7
+ARG GPERFTOOLS_PATH=/usr/local/gperftools
+RUN wget -q https://github.com/gperftools/gperftools/releases/download/gperftools-${GPERFTOOLS_VERSION}/gperftools-${GPERFTOOLS_VERSION}.tar.gz -O gperftools.tar.gz && \
+    tar -xzf gperftools.tar.gz && \
+    cd gperftools-${GPERFTOOLS_VERSION} && \
+    ./configure \
+      --prefix=${GPERFTOOLS_PATH} && \
+    make -j$(nproc) && \
+    make install && \
+    rm -rf /root/gperftools.tar.gz /root/gperftools-${GPERFTOOLS_VERSION}
+
+# Install HPX
+ARG HPX_MAJOR=1
+ARG HPX_MINOR=4
+ARG HPX_PATCH=1
+ARG HPX_PATH=/usr/local/hpx
+RUN wget -q https://github.com/STEllAR-GROUP/hpx/archive/${HPX_MAJOR}.${HPX_MINOR}.${HPX_PATCH}.tar.gz -O hpx.tar.gz && \
+    tar -xzf hpx.tar.gz && \
+    cd hpx-${HPX_MAJOR}.${HPX_MINOR}.${HPX_PATCH} && \
+    mkdir build && \
+    cd build && \
+    cmake .. \
+      -DBOOST_ROOT=$BOOST_PATH \
+      -DHWLOC_ROOT=$HWLOC_PATH \
+      -DTCMALLOC_ROOT=$GPERFTOOLS_PATH \
```

Comment:
Have you checked if you get the expected configuration? I would check if `HPX_WITH_MALLOC` has to be set manually or it is automatically detected.

(In case, see also `build-env-debug.Dockerfile`)

Reply:
Seems like it is picked up, from the logs it seems `libhwloc.so.15.2.0` is linked against :)